average.js

```diff
@@ -1,6 +1,12 @@
-
 function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+    //filtering for NaN
+    const filterNaN = numbers.filter(value => !Number.isNaN(value));
+    let maxValue = Math.max(...numbers);
+    let minValue = Math.min(...numbers);
+    let variation = maxValue - minValue;
+    // variation is computed as maximum number - minimum number
+    if (variation > 0.95 * maxValue) return NaN;
```

Comment:
Good an clean solution :+1: 
Any reason / reference to choose 0.95 as the factor to judge variability?

Reply:
I assumed that the readings would be of **Gaussian** distribution. Then I considered a high confidence level (95%) to be sure that there is error in the readings. Spread of the curve can be approximated as Max value - min value. A high spread (95% of max value) will indicate high chances of error.

Although this approach is very naïve, false positive errors would be reduced. This can be further improved by using variance. 

https://files.eric.ed.gov/fulltext/EJ1052989.pdf